### PR TITLE
Product: Migrate media library endpoint to wp/v2 to support sites without XMLRPC

### PR DESCRIPTION
**Pull request by itsmeichigo**
Part of #14550 

## Description

Previously, we used the `rest/v1.1/sites/<siteID>/media` endpoint to retrieve the media library for sites connected to the Jetpack plugin. This doesn't work when the site has XMLRPC disabled.

This PR fixes that by unifying the logic and using `wp/v2/site/<siteID>/media` for all sites disregarding how the user is authenticated in the app. This ensures that XMLRPC is no longer required for loading the media library.

## Steps to reproduce

**a. No Jetpack case (Application Password)**
1. Have a .org site with no Jetpack, 
2. Install "Disable XML-RPC-API" plugin.
3. In the app, login to site with application password,
4. Go to Products > Add Product > Add Product Image 
5. Choose WordPress Media Library option as the source for uploading photo
6. Confirm that images in the media library are loaded successfully.

**b. Jetpack CP site case**
1. Have a .org site with no Jetpack but with a plugin with Jetpack CP (I use "Blaze Ads" and connect WPCOM account while attempting to create a Blaze campaign from the `wp-admin` page)
2. Install "Disable XML-RPC-API" plugin.
3. Setup the Jetpack CP plugin and connect it to your WordPress.com account
4. In the app, login to site with WordPress.com account,
5. Go to Products > Add Product > Add Product Image 
6. Choose WordPress Media Library option as the source for uploading photo
7. Confirm that images in the media library are loaded successfully.

**c. WordPress.com / full Jetpack site case**
1. Have a .org site with Jetpack plugin
2. Install "Disable XML-RPC-API" plugin.
3. Setup the Jetpack plugin and connect it to your WordPress.com account,
4. In the app, login to site with WordPress.com account,
5. Go to Products > Add Product > Add Product Image 
6. Choose WordPress Media Library option as the source for uploading photo
7. Confirm that images in the media library are loaded successfully.

## Testing information

Tested on simulator iPhone 16 Pro iOS 18.1 and confirmed that media library loads successfully for sites without XMLRPC (authenticated with/without WPCom).

## Screenshots

N/A

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [ ] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [ ] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [ ] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on all devices (phone/tablet) and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Networking/NetworkingTests/Responses/media-library-from-wordpress-site.json
Comment (selanthiraiyan) [inline Networking/NetworkingTests/Responses/media-library-from-wordpress-site.json:1]: nit: Should we rename this file to `media-library.json` as the response can come from self-hosted non-WPCOM sites as well?

File: Yosemite/YosemiteTests/Mocks/Networking/Remote/MockMediaRemote.swift [hunk 1]
```diff
@@ -12,11 +12,8 @@ final class MockMediaRemote {
     /// The results to return based on the given site ID in `loadMedia`
     private var loadMediaResultsBySiteID = [Int64: Result<WordPressMedia, Error>]()
 
-    /// The results to return based on the given site ID in `loadMediaLibrary`
-    private var loadMediaLibraryResultsBySiteID = [Int64: Result<[Media], Error>]()
-
     /// The results to return based on the given site ID in `loadMediaLibraryFromWordPressSite`
```
Comment (selanthiraiyan) [inline Yosemite/YosemiteTests/Mocks/Networking/Remote/MockMediaRemote.swift]: Nit: The comment needs to be renamed as well.

**Review by selanthiraiyan — APPROVED** [review]
Thanks for working on this, Huong!

I confirmed that media library can be loaded with XML-RPC disabled on JCP, Jetpack and non-Jetpack sites. Code looks good to me. 

🚀

File: Networking/NetworkingTests/Responses/media-library-from-wordpress-site.json
Comment (itsmeichigo) [reply Networking/NetworkingTests/Responses/media-library-from-wordpress-site.json:1]: Done in b4e5bd680cd455ecaa10d4541b1ce9b01c8c28be.

File: Yosemite/YosemiteTests/Mocks/Networking/Remote/MockMediaRemote.swift (hunk 1 above)
Comment (itsmeichigo) [reply Yosemite/YosemiteTests/Mocks/Networking/Remote/MockMediaRemote.swift]: Done in 79a826e6f99f5bb5a61bdd5ba77a53bd6c7d7ee8.

**Pull request merged**